x-pack/plugins/siem/public/components/page/hosts/uncommon_process_table/index.tsx

```diff
@@ -221,49 +170,21 @@ const getUncommonColumns = (): Array<Columns<UncommonProcessesEdges>> => [
     name: i18n.HOSTS,
     truncateText: false,
     hideForMobile: false,
-    render: ({ node }) => {
-      const hosts: HostEcsFields[] = node.host;
-      const draggables = hosts
-        .filter(({ name }) => name != null && name[0] != null)
-        .map(({ name }, index) => {
-          if (name != null && name[0] != null) {
-            const dwId = escapeDataProviderId(
-              `uncommon-process-table-${node._id}-hostName-${name[0]}`
-            );
-            return (
-              <React.Fragment key={dwId}>
-                {index !== 0 ? <>,&nbsp;</> : null}
-                <DraggableWrapper
-                  key={dwId}
-                  dataProvider={{
-                    and: [],
-                    enabled: true,
-                    id: dwId,
-                    name: name[0],
-                    excluded: false,
-                    kqlQuery: '',
-                    queryMatch: {
-                      field: 'host.name',
-                      value: name[0],
-                    },
-                  }}
-                  render={(dataProvider, _, snapshot) =>
-                    snapshot.isDragging ? (
-                      <DragEffects>
-                        <Provider dataProvider={dataProvider} />
-                      </DragEffects>
-                    ) : (
-                      <HostDetailsLink hostName={name[0]}>{name}</HostDetailsLink>
-                    )
-                  }
-                />
-              </React.Fragment>
-            );
-          } else {
-            return getEmptyTagValue();
-          }
-        });
-      return draggables.length > 0 ? draggables : getEmptyTagValue();
-    },
+    render: ({ node }) =>
+      getRowItemDraggables({
+        rowItems: getHostNames(node),
+        attrName: 'host.name',
+        idPrefix: `uncommon-process-table-${node._id}-processHost`,
+      }),
   },
 ];
+
+export const getHostNames = (node: UncommonProcessItem): string[] => {
+  if (node.host != null) {
+    return node.host
+      .filter(host => host.name != null && host.name[0] != null)
+      .map(host => (host.name != null && host.name[0] != null ? host.name[0] : ''));
```

Comment:
I wonder if we should rename `UncommonProcessItem.host` to be `UncommonProcessItem.hosts` since it's an array of hosts. Also, is it necessary to check for null in the `map` as well since they're removed in the previous `filter`?

Reply:
I wish it wasn't necessary. (I am looking for a better way). I wish I could cast it out as a required field from the filter field to the map field and then not check for null again.

Reply:
Yea, I can change it from `host` to `hosts`. I think that would be clearer